src/js/msp.js

```diff
@@ -193,8 +288,40 @@ var MSP = {
             bufView[4] = code; // code
             bufView[5] = bufView[3] ^ bufView[4]; // checksum
         }
+        return bufferOut;
+    },
+    encode_message_v2: function (code, data) {
+        const dataLength = data ? data.length : 0;
+        // 9 bytes for protocol overhead
+        const bufferSize = dataLength + 9;
+        const bufferOut = new ArrayBuffer(bufferSize);
+        const bufView = new Uint8Array(bufferOut);
+        bufView[0] = 36; // $
+        bufView[1] = 88; // X
+        bufView[2] = 60; // <
+        bufView[3] = 0;  // flag
+        bufView[4] = code & 0xFF;
+        bufView[5] = (code >> 8) & 0xFF;
+        bufView[6] = dataLength & 0xFF;
+        bufView[7] = (dataLength >> 8) & 0xFF;
+        for (let ii = 0; ii < dataLength; ii++) {
+            bufView[8 + ii] = data[ii];
+        }
+        bufView[bufferSize - 1] = this.crc8_dvb_s2_data(bufView, 3, bufferSize - 1);
+        return bufferOut;
+    },
+    send_message: function (code, data, callback_sent, callback_msp, doCallbackOnError) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A function should not mix return statements with and without a result.](https://app.codacy.com/gh/emuflight/EmuConfigurator/pullRequest?prid=5635143)